lib/database/utility.php

```diff
@@ -624,6 +626,20 @@ function SubmitCodeNote2($user, $gameID, $address, $note)
         return false;
     }
 
+    $addressHex = '0x' . str_pad(dechex($address), 6, '0', STR_PAD_LEFT);
+    $currentNotes = getCodeNotesData($gameID);
+    $i = array_search($addressHex, array_column($currentNotes, 'Address'));
+
+    if (
+        $i === false

```

Comment:
doesn't `array_search` return false if the item is _not_ found?

Reply:
yep, and if item is not found that address is available for a regular user to edit.

Reply:
seems like your logic is backwards. combined with the following lines, I'm reading:

"if not found && not a developer && not the current user && not empty"

shouldn't it be

"if found && not a developer && not the current user && not empty"?

Reply:
oh sh!t, you're right! it should be `$i !== false`.